### PR TITLE
Don't serialize team if it is None

### LINKED ISSUE #808: String "null" sent as team in game_info
If the server has received info that a player has joined the game it could potentially send "null" since it might not have received the Team info for the player yet

### PR DESCRIPTION
**Pull request by Sheikah45**
Fixes #808

### CODE REVIEW

File: server/games/game.py [hunk 1]
```diff
@@ -908,7 +908,7 @@ def to_dict(self):
                     player.login for player in self.players
                     if self.get_player_option(player.id, "Team") == team
                 ]
-                for team in self.teams
+                for team in self.teams if team
```
Comment (Askaholic) [inline server/games/game.py:911]: Maybe do `is None` since this will also ignore the team if it’s set to `0`

Comment (Sheikah45) [reply server/games/game.py:911]: Yeah I can do that but as far as I know 0 would be an invalid value as well. But still would be good.

Comment (Askaholic) [reply server/games/game.py]: Yea it shouldn’t come up in SupCom, but still nice to be robust. Also there is TAF which forked from here and occasionally merges our new changes in, and in TA maybe `0` is valid idk.

File: server/games/game.py [hunk 2]
```diff
@@ -908,7 +908,7 @@ def to_dict(self):
                     player.login for player in self.players
                     if self.get_player_option(player.id, "Team") == team
                 ]
-                for team in self.teams
+                for team in self.teams if team is None
```
Comment (Askaholic) [inline server/games/game.py:911]: My bad `is not None` :P

Comment (Sheikah45) [reply server/games/game.py:911]: Yeah I should have realized ha

**Review by Askaholic — APPROVED** [review]

**Pull request merged**